hawtio-system/src/main/java/io/hawt/web/AuthenticationFilter.java

```diff
@@ -79,10 +79,10 @@ public void init(FilterConfig filterConfig) throws ServletException {
 
         // JVM system properties can override always
         if (System.getProperty(HAWTIO_AUTHENTICATION_ENABLED) != null) {
-            configuration.setEnabled(Boolean.getBoolean(HAWTIO_AUTHENTICATION_ENABLED));
+            configuration.setEnabled(Boolean.getBoolean(System.getProperty(HAWTIO_AUTHENTICATION_ENABLED)));
         }
         if (System.getProperty(HAWTIO_NO_CREDENTIALS_401) != null) {
-            configuration.setNoCredentials401(Boolean.getBoolean(HAWTIO_NO_CREDENTIALS_401));
+            configuration.setNoCredentials401(Boolean.getBoolean(System.getProperty(HAWTIO_NO_CREDENTIALS_401)));
```

Comment:
This is not really a bug. Instead, the proposed one has a bug. See the javadoc for `Boolean#getBoolean()`.